pkg/project/resource_project_repo_test.go

```diff
@@ -114,3 +116,127 @@ func TestAccProjectRepo(t *testing.T) {
 		},
 	})
 }
+
+/*
+Test to assign large number of repositories to a project
+*/
+func TestAccAssignMultipleReposInProject(t *testing.T) {
+
+	const numRepos = 100
+	const repoNameInitial = "repo-"
+
+	name := "tftestprojects" + randSeq(10)
+	resourceName := "project." + name
+	projectKey := strings.ToLower(randSeq(6))
+
+	repoNamesStr := func(repoCount int) string {
+		var repoNames []string
+		for i := 0; i < repoCount; i++ {
+			repoNames = append(repoNames, fmt.Sprintf("%s%d", repoNameInitial, i))
+		}
+		jsonStr, err := json.Marshal(repoNames)
+		if err != nil {
+			return "[]"
+		}
+		return string(jsonStr)
+	}
+
+	preCheck := func(t *testing.T, numRepo int) func() {
+		return func() {
+			testAccPreCheck(t)
+			for i := 0; i < numRepo; i++ {
+				createTestRepo(t, fmt.Sprintf("%s%d", repoNameInitial, i))
```

Comment:
So, the problem with creating resources in a predictable way like this is that if they don't get destroyed during clean up, the whole platform instance will likely need to be destroyed since subsequent test runs will have naming collisions.

There is a reason the other tests in RT use randomized naming